services/director-v2/src/simcore_service_director_v2/modules/db/repositories/project_networks.py

```diff
@@ -0,0 +1,41 @@
+import json
+
+import sqlalchemy as sa
+from aiopg.sa.result import RowProxy
+from models_library.project_networks import NetworksWithAliases, ProjectNetworks
+from models_library.projects import ProjectID
+from sqlalchemy.dialects.postgresql import insert as pg_insert
+
+from ....core.errors import ProjectNotFoundError
+from ..tables import project_networks
+from ._base import BaseRepository
+
+
+class ProjectNetworksRepository(BaseRepository):
+    async def get_project_networks(self, project_id: ProjectID) -> ProjectNetworks:
+        async with self.db_engine.acquire() as conn:
+            row: RowProxy = await (
+                await conn.execute(
+                    sa.select([project_networks]).where(
+                        project_networks.c.project_uuid == f"{project_id}"
+                    )
+                )
+            ).first()
+        if not row:
+            raise ProjectNotFoundError(project_id)
```

Comment:
nice. An interface is not only about the functions and signatures it exposes but also about the exceptions it raises.